### DISCUSSION TITLE
Is there a way to include aromatic or aliphatic tags for MCSResult?

### DISCUSSION THREAD

**Discussion by dangthatsright** [rdkit/rdkit#6118] [Q&A] [answered]
I wasn't able to find anything in my search but I figured it would be a common use case?

When using FindMCS, you can specify `bondCompare=rdFMCS.BondCompare.CompareOrderExact` to get the proper bond types in the output query mol. Is there an equivalent way to get more specific atom types in the output query mol? Currently `atomCompare` can only use `rdFMCS.AtomCompare.CompareElements` as the most specific comparator but the atom matching 
 it gives is only `[#6]` whereas I'm looking for `[c]` or `[C]` if possible. 

Perhaps a  custom `AtomComparator` implementation would work? but I don't couldn't find any examples on how to use this

Thanks for any help!

**Comment by greglandrum** [comment, marked as answer]
@dangthatsright You can't change what shows up in the SMARTS that the MCS code generates, but you can change the atom comparison function so that aromatic atoms only match other aromatic atoms and then match the resulting SMARTS back to the molecule in order to get a SMILES with aromatic/aliphatic atoms.

Here's a gist which shows how to do this:
https://gist.github.com/greglandrum/c1ac38894694d6df6e3880ebaddc2963

**Reply by dangthatsright** [reply]
thanks for the workaround and example!

**Reply by jrkarki** [reply]
Thanks @greglandrum. This was the solution I was looking for. This also resolves my issue (https://github.com/rdkit/rdkit/discussions/6031) I posted in the discussions group a while ago. 
I will add the solution there and marked it as solved.
